core/src/com/unciv/models/translations/TranslationFileWriter.kt

```diff
@@ -17,99 +18,130 @@ import java.lang.reflect.Field
-                    languageHashmap[translation.entry] = translation[language]!!
-            }
-            writeByTemplate(language, languageHashmap)
-        }
-        writeLanguagePercentages(translations)
+        return countOfTranslatedLines
     }
 
-    private fun writeLanguagePercentages(translations: Translations){
-        val percentages = translations.calculatePercentageCompleteOfLanguages()
+    private fun writeLanguagePercentages(percentages: HashMap<String,Int>){
         val stringBuilder = StringBuilder()
         for(entry in percentages){
             stringBuilder.appendln(entry.key+" = "+entry.value)
         }
         Gdx.files.local(TranslationFileReader.percentagesFileLocation).writeString(stringBuilder.toString(),false)
     }
 
-    private fun generateTutorialsStrings(): Collection<String> {
 
-        if (generatedStrings.containsKey("Tutorials"))
-            return generatedStrings.getValue("Tutorials")
 
-        generatedStrings["Tutorials"] = mutableSetOf()
-        val tutorialsStrings = generatedStrings["Tutorials"]
+    private fun generateTutorialsStrings(): MutableSet<String> {
+
+        val tutorialsStrings = mutableSetOf<String>()
         val tutorials = JsonParser().getFromJson(LinkedHashMap<String, Array<String>>().javaClass, "jsons/Tutorials.json")
 
+        var uniqueIndexOfNewLine = 0
         for (tutorial in tutorials) {
             for (str in tutorial.value)
-                if (str != "") tutorialsStrings!!.add("$str = ")
+                if (str != "") tutorialsStrings.add("$str = ")
+            // This is a small hack to insert multiple /n into the set, which can't contain identical lines
+            tutorialsStrings.add("$specialNewLineCode ${uniqueIndexOfNewLine++}")
         }
-        return tutorialsStrings!!
+        return tutorialsStrings
     }
 
+    // used for unit test only
     fun getGeneratedStringsSize(): Int {
-        if (generatedStrings.isEmpty())
-            generateStringsFromJSONs()
-        return generatedStrings.values.sumBy { it.size }
+        return generateStringsFromJSONs().values.sumBy { // exclude empty lines
+            it.count{ line: String -> !line.startsWith(specialNewLineCode) } }
     }
 
-    private fun generateStringsFromJSONs() {
+    private fun generateStringsFromJSONs(modFolder: FileHandle? = null): Map<String, MutableSet<String>> {
 
-        if (generatedStrings.isNotEmpty())
-            return // do not regenerate if the strings are ready
+        // Using LinkedHashMap (instead of HashMap) is important to maintain the order of sections in the translation file
+        val generatedStrings = LinkedHashMap<String, MutableSet<String>>()
 
+        var uniqueIndexOfNewLine = 0
         val jsonParser = JsonParser()
-        val folderHandler = Gdx.files.internal("jsons")
+        val folderHandler = if (modFolder != null) modFolder.child("jsons")
+                             else Gdx.files.internal("jsons")
```

Comment:
I notice this is the third time for this pattern - may I suggest fun getLocation(subPath:String, modFolder:String?) = if(modfolder=null) modFolder.child(subPath) else Gdx.files.internal(subPath)

Reply:
Good suggestion, thank you.